gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/portal_page/PortalNavigationItemsQueryServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+
+    public PortalNavigationItemsQueryServiceImpl(@Lazy final PortalNavigationItemRepository portalNavigationItemRepository) {
+        this.portalNavigationItemRepository = portalNavigationItemRepository;
+    }
+
+    @Override
+    public PortalNavigationItem findByIdAndEnvironmentId(String environmentId, PortalNavigationItemId id) {
+        try {
+            var result = portalNavigationItemRepository.findById(id.json());
+            if (result.isPresent() && result.get().getEnvironmentId().equals(environmentId)) {
+                return portalNavigationItemAdapter.toEntity(result.get());
+            }
+            return null;
+        } catch (TechnicalException e) {
+            String errorMessage = String.format(
+                "An error occurred while finding portal navigation item by id %s and environmentId %s",
+                id,
+                environmentId
+            );
+            logger.error(errorMessage, e);
+            throw new TechnicalDomainException(errorMessage, e);
+        }
+    }
+
+    @Override
+    public Collection<PortalNavigationItem> findByParentIdAndEnvironmentId(String environmentId, PortalNavigationItemId parentId) {
+        try {
+            var results = portalNavigationItemRepository.findAllByParentIdAndEnvironmentId(parentId.json(), environmentId);
+            return results.stream().map(portalNavigationItemAdapter::toEntity).collect(Collectors.toList());
+        } catch (TechnicalException e) {
+            String errorMessage = String.format(
+                "An error occurred while finding portal navigation items by parentId %s and environmentId %s",
+                parentId,
+                environmentId
+            );
+            logger.error(errorMessage, e);
+            throw new TechnicalDomainException(errorMessage, e);
+        }
+    }
+
+    @Override
+    public Collection<PortalNavigationItem> findTopLevelItemsByEnvironmentId(String environmentId, PortalArea portalArea) {
+        try {
+            var area = switch (portalArea) {
+                case HOMEPAGE -> io.gravitee.repository.management.model.PortalNavigationItem.Area.HOMEPAGE;
+                case TOP_NAVBAR -> io.gravitee.repository.management.model.PortalNavigationItem.Area.TOP_NAVBAR;
+            };
+            var results = portalNavigationItemRepository.findAllByAreaAndEnvironmentIdAndParentIdIsNull(area, environmentId);
+            return results.stream().map(portalNavigationItemAdapter::toEntity).collect(Collectors.toList());
+        } catch (TechnicalException e) {
+            String errorMessage = String.format(
+                "An error occurred while finding top level portal navigation items by environmentId %s and area %s",
+                environmentId,
+                portalArea
+            );
+            logger.error(errorMessage, e);
+            throw new TechnicalDomainException(errorMessage, e);
+        }
```

Comment:
don't log & throw.
=> https://gravitee.slab.com/posts/good-practices-for-handling-java-exceptions-and-logging-1dc30qlp